routers/web/repo/view.go

```diff
@@ -127,12 +173,20 @@ func renderDirectory(ctx *context.Context, treeLink string) {
 		ctx.Data["CanUploadFile"] = setting.Repository.Upload.Enabled && !ctx.Repo.Repository.IsArchived
 	}
 
-	readmeFile, readmeTreelink := findReadmeFile(ctx, entries, treeLink)
-	if ctx.Written() || readmeFile == nil {
+	if ctx.Written() {
+		return
+	}
+
+	readmeFile, err := findReadmeFileInEntries(ctx, entries)
+	if err != nil {
+		ctx.ServerError("findReadmeFileInEntries", err)
+		return
+	}
+	if readmeFile == nil {
 		return
 	}
 
-	renderReadmeFile(ctx, readmeFile, readmeTreelink)
+	renderReadmeFile(ctx, readmeFile, treeLink)
```

Comment:
The README file maybe in a sub folder of `treeLink`.

Reply:
I'm not sure what you're getting at @lunny. The code works equally when the README is in a subfolder or at the top:

![Screenshot 2023-01-26 at 13-04-06 readme-test](https://user-images.githubusercontent.com/987487/214914357-e6de85f8-5d93-43a7-ba2c-656a5542aea4.png)
![Screenshot 2023-01-26 at 13-04-29 readme-test](https://user-images.githubusercontent.com/987487/214914358-82fc84a3-f301-49db-af50-2cf2b13dfbc6.png)

`readmeFile.name` includes the rest of the relative path:

https://github.com/go-gitea/gitea/blob/bd43cc2e14ea9486a7a2296bd20031448e4832d7/routers/web/repo/view.go#L150

which is then tacked onto `treeLink` over in

https://github.com/go-gitea/gitea/blob/bd43cc2e14ea9486a7a2296bd20031448e4832d7/routers/web/repo/view.go#L315-L316

This is the best way I can see to do this, since `findReadmeFileInEntries()` is the place where the subfolder is detected if it exists. I suppose I could extend `type namedBlob struct` with an extra `subfolder string` instead of prepending it, but that would be more code for little benefit IMO.

I could change the variable name

```suggestion
	renderReadmeFile(ctx, readmeFile, readmeTreelink)
```

to make this function consistent with `renderReadmeFile()` -- or vice versa -- if you like, but that's just cosmetic.

Reply:
Hi @lunny. Happy new year again! Does my explanation make sense to you? I am afraid I might have misunderstood your feedback for a second time.